lib/Slackbot_worker.js

```diff
@@ -86,10 +105,22 @@ module.exports = function(botkit, config) {
             bot.rtm = new Ws(res.url, null, {agent: agent});
             bot.msgcount = 1;
 
-            var pingIntervalId = null;
+            bot.rtm.on('pong', function (obj) {
+                lastPong = Date.now()
+                botkit.debug('PONG received');
+            })
+
             bot.rtm.on('open', function() {
 
                 pingIntervalId = setInterval(function() {
+                    if (lastPong && lastPong + 12000 < Date.now()) {
+                        var err = new Error('Stale RTM connection, closing RTM')
+                        botkit.log.error(err);
```

Comment:
Unified error handling and passing to log system is mandatory for better maintenance. Therefore, please update this line like line 154. `botkit.log.error('RTM websocket error!', err);`